libevmasm/Assembly.cpp

```diff
@@ -891,6 +891,18 @@ LinkerObject const& Assembly::assemble() const
 	std::multimap<size_t, size_t> subRef;
 	std::vector<unsigned> sizeRef; ///< Pointers to code locations where the size of the program is inserted
 	unsigned bytesPerTag = numberEncodingSize(bytesRequiredForCode);
+	// Adjust bytesPerTag for references to sub assemblies.
+	for (AssemblyItem const& i: m_items)
+		if (i.type() == PushTag)
+		{
+			auto [subId, tagId] = i.splitForeignPushTag();
+			if (subId == std::numeric_limits<size_t>::max())
+				continue;
+			assertThrow(subId < m_subs.size(), AssemblyException, "Invalid sub id");
+			for (auto subTagPosition: m_subs[subId]->m_tagPositionsInBytecode)
+				if (subTagPosition != std::numeric_limits<size_t>::max())
+					bytesPerTag = std::max(bytesPerTag, numberEncodingSize(subTagPosition));
+		}
```

Comment:
For the record: we need to adjust ``bytesPerTag`` to not only be sufficiently large to hold any tag into the current sources - but we also need to choose it large enough for tags to subassemblies (we use the same size for all tags - we could do something more clever in general, but that'd be a separate improvement).

Without this, some cases that actually contained references to subassembly tags of size 2 from an assembly with tags of size 1 in fact failed.